### PR TITLE
Add support for Results extension point

### LINKED ISSUE #35508: Allow external libraries to extend Results
When using minimal APIs, external libraries could want to expose custom results, like views. There is no way to add custom extension methods to the `Results` class. However a solution could be to expose a public property e.g. `Results.Custom` that could be used to expose other types of results.

Today the only solution is for each library to create their own factory class, or return an `IResult` instance directly which doesn't help for discoverability, and consistency.

Example:
```c#
app.MapGet("/", () =>
{
    return LiquidResults.View("index", new Todo(1, "Go back to work!", false));
});
```
(https://github.com/sebastienros/fluid/pull/292/files#diff-b2a1bda3022b9e2130cc1ececbfc576ec1f91dca2c65baef80c1604c8c36821eR11-R14)

Which could instead become:
```c#
app.MapGet("/", () =>
{
    return Results.Custom.LiquidView("index", new Todo(1, "Go back to work!", false));
});
```

### PR DESCRIPTION
**Pull request by captainsafia**
Fixes https://github.com/dotnet/aspnetcore/issues/35508.

See https://github.com/dotnet/aspnetcore/issues/35508#issuecomment-902873299 for more info.

### CODE REVIEW

File: src/Http/Http.Results/src/IResultExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Result
+{
+    /// <summary>
+    /// Provides an interface to registering external methods that provide
```
Comment (DamianEdwards) [inline src/Http/Http.Results/src/IResultExtensions.cs:7]:
```suggestion
    /// Provides an interface for registering external methods that provide
```

File: src/Http/Http.Results/src/IResultExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    /// custom IResult instances.
```
Comment (halter73) [inline src/Http/Http.Results/src/IResultExtensions.cs:8]:
```suggestion
    /// custom <see cref="IResult"/> instances.
```

File: src/Http/Http.Results/src/ResultExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Result
+{
+    /// <summary>
+    /// Implements an interface for registering external methods that provide
+    /// custom IResult instances.
```
Comment (halter73) [inline src/Http/Http.Results/src/ResultExtensions.cs:8]:
```suggestion
    /// custom <see cref="IResult"/> instances.
```

File: src/Http/Http.Results/src/Results.cs [hunk 4]
```diff
@@ -588,5 +588,11 @@ public static IResult Accepted(string? uri = null, object? value = null)
         /// <returns>The created <see cref="IResult"/> for the response.</returns>
         public static IResult AcceptedAtRoute(string? routeName = null, object? routeValues = null, object? value = null)
             => new AcceptedAtRouteResult(routeName, routeValues, value);
+
+        /// <summary>
+        /// Provides a container for external libraries to extend
+        /// the default `Results` set with their own samples.
```
Comment (halter73) [inline src/Http/Http.Results/src/Results.cs:594]:
```suggestion
        /// the default <see cref="Results"/> set with their own samples.
```

**Review by halter73 — APPROVED** [review]

File: src/Http/Http.Results/src/Results.cs (hunk 4 above)
Comment (DamianEdwards) [inline src/Http/Http.Results/src/Results.cs:594]:
```suggestion
        /// Provides a place for external libraries to extend the default `Results` set
        /// via extension methods returning custom <see cref="IResult" /> implementations.
```

File: src/Http/Http.Results/src/IResultExtensions.cs (hunk 2 above)
Comment (DamianEdwards) [inline src/Http/Http.Results/src/IResultExtensions.cs:8]:
```suggestion
    /// Provides an interface for registering external methods that provide
    /// custom <see cref="IResult" /> instances.
```

File: src/Http/Http.Results/src/Results.cs [hunk 5]
```diff
@@ -588,5 +588,11 @@ public static IResult Accepted(string? uri = null, object? value = null)
         /// <returns>The created <see cref="IResult"/> for the response.</returns>
         public static IResult AcceptedAtRoute(string? routeName = null, object? routeValues = null, object? value = null)
             => new AcceptedAtRouteResult(routeName, routeValues, value);
+
+        /// <summary>
+        /// Provides a place for external libraries to extend the default `Results` set
```
Comment (halter73) [inline src/Http/Http.Results/src/Results.cs:593]:
```suggestion
        /// Provides a place for external libraries to extend the default <see cref="Results"/> set
```

Does this not work?

Comment (captainsafia) [reply src/Http/Http.Results/src/Results.cs:593]: It does. Apparently, GH only lets you pick one suggestion against a single diff so I picked Damian's. I thought it had this change but looks like it didn't.

File: src/Http/Http.Results/src/ResultExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Result
+{
+    /// <summary>
+    /// Implements an interface for registering external methods that provide
+    /// custom <see cref="IResult"/> instances.
+    /// </summary>
+    internal class ResultExtensions : IResultExtensions { }
```
Comment (Kahbazi) [inline src/Http/Http.Results/src/ResultExtensions.cs:10]:
```suggestion
    internal sealed class ResultExtensions : IResultExtensions { }
```

**Pull request merged**